File: frontend/src/data/seed.ts
import type { Driver, Expense, FuelLog, MaintenanceLog, Trip, User, Vehicle } from '../types'

export const seedUsers: User[] = [
  { id: 'u1', name: 'Maria Chen', email: 'manager@transitops.com', password: 'password123', role: 'Fleet Manager' },
  { id: 'u2', name: 'Alex Rivera', email: 'driver@transitops.com', password: 'password123', role: 'Driver' },
  { id: 'u3', name: 'Sam Okafor', email: 'safety@transitops.com', password: 'password123', role: 'Safety Officer' },
  { id: 'u4', name: 'Priya Nair', email: 'finance@transitops.com', password: 'password123', role: 'Financial Analyst' },
]

export const seedVehicles: Vehicle[] = [
  { id: 'v1', registrationNumber: 'VAN-05', model: 'Ford Transit', type: 'Van', maxLoadCapacityKg: 500, odometerKm: 42150, acquisitionCost: 38000, status: 'Available', region: 'North' },
  { id: 'v2', registrationNumber: 'TRK-12', model: 'Isuzu NPR', type: 'Truck', maxLoadCapacityKg: 3500, odometerKm: 88320, acquisitionCost: 62000, status: 'On Trip', region: 'South' },
  { id: 'v3', registrationNumber: 'VAN-08', model: 'Mercedes Sprinter', type: 'Van', maxLoadCapacityKg: 800, odometerKm: 15200, acquisitionCost: 45000, status: 'In Shop', region: 'East' },
  { id: 'v4', registrationNumber: 'TRK-03', model: 'Volvo FL', type: 'Truck', maxLoadCapacityKg: 5000, odometerKm: 121000, acquisitionCost: 89000, status: 'Retired', region: 'West' },
  { id: 'v5', registrationNumber: 'VAN-11', model: 'Ford Transit', type: 'Van', maxLoadCapacityKg: 500, odometerKm: 9800, acquisitionCost: 39500, status: 'Available', region: 'North' },
  { id: 'v6', registrationNumber: 'TRK-19', model: 'Isuzu NPR', type: 'Truck', maxLoadCapacityKg: 3500, odometerKm: 55400, acquisitionCost: 63000, status: 'On Trip', region: 'South' },
  { id: 'v7', registrationNumber: 'CAR-02', model: 'Toyota Corolla', type: 'Car', maxLoadCapacityKg: 150, odometerKm: 31000, acquisitionCost: 21000, status: 'Available', region: 'East' },
]

export const seedDrivers: Driver[] = [
  { id: 'd1', name: 'Alex Rivera', licenseNumber: 'LIC-10023', licenseCategory: 'C', licenseExpiry: '2027-03-15', contact: '555-0101', safetyScore: 92, status: 'Available' },
  { id: 'd2', name: 'Jordan Blake', licenseNumber: 'LIC-10078', licenseCategory: 'C+E', licenseExpiry: '2026-11-02', contact: '555-0102', safetyScore: 87, status: 'On Trip' },
  { id: 'd3', name: 'Taylor Reed', licenseNumber: 'LIC-10099', licenseCategory: 'B', licenseExpiry: '2026-01-20', contact: '555-0103', safetyScore: 78, status: 'Off Duty' },
  { id: 'd4', name: 'Morgan Diaz', licenseNumber: 'LIC-10112', licenseCategory: 'C', licenseExpiry: '2025-09-01', contact: '555-0104', safetyScore: 65, status: 'Suspended' },
  { id: 'd5', name: 'Casey Kim', licenseNumber: 'LIC-10145', licenseCategory: 'C+E', licenseExpiry: '2027-06-30', contact: '555-0105', safetyScore: 95, status: 'On Trip' },
  { id: 'd6', name: 'Riley Santos', licenseNumber: 'LIC-10201', licenseCategory: 'B', licenseExpiry: '2026-08-14', contact: '555-0106', safetyScore: 89, status: 'Available' },
]

export const seedTrips: Trip[] = [
  { id: 't1', source: 'Warehouse A', destination: 'Downtown Depot', vehicleId: 'v2', driverId: 'd2', cargoWeightKg: 2800, plannedDistanceKm: 65, status: 'Dispatched', createdAt: '2026-07-10T08:00:00Z', dispatchedAt: '2026-07-10T08:15:00Z' },
  { id: 't2', source: 'Port Terminal', destination: 'North Storage', vehicleId: 'v6', driverId: 'd5', cargoWeightKg: 3100, plannedDistanceKm: 40, status: 'Dispatched', createdAt: '2026-07-11T06:30:00Z', dispatchedAt: '2026-07-11T07:00:00Z' },
  { id: 't3', source: 'Warehouse B', destination: 'Retail Center', vehicleId: 'v1', driverId: 'd1', cargoWeightKg: 420, plannedDistanceKm: 22, actualDistanceKm: 23.5, fuelConsumedLiters: 6.1, status: 'Completed', createdAt: '2026-07-08T09:00:00Z', dispatchedAt: '2026-07-08T09:20:00Z', completedAt: '2026-07-08T11:05:00Z' },
  { id: 't4', source: 'Depot 5', destination: 'Customer Site', vehicleId: 'v5', driverId: 'd6', cargoWeightKg: 300, plannedDistanceKm: 18, status: 'Draft', createdAt: '2026-07-12T07:00:00Z' },
]

export const seedMaintenanceLogs: MaintenanceLog[] = [
  { id: 'm1', vehicleId: 'v3', description: 'Brake pad replacement', cost: 420, openedAt: '2026-07-09T10:00:00Z', status: 'Active' },
  { id: 'm2', vehicleId: 'v1', description: 'Oil change', cost: 90, openedAt: '2026-06-15T10:00:00Z', closedAt: '2026-06-15T14:00:00Z', status: 'Closed' },
]

export const seedFuelLogs: FuelLog[] = [
  { id: 'f1', vehicleId: 'v1', liters: 40, cost: 62, date: '2026-07-05' },
  { id: 'f2', vehicleId: 'v2', liters: 120, cost: 186, date: '2026-07-08' },
  { id: 'f3', vehicleId: 'v6', liters: 95, cost: 148, date: '2026-07-09' },
  { id: 'f4', vehicleId: 'v1', liters: 38, cost: 59, date: '2026-06-20' },
]

export const seedExpenses: Expense[] = [
  { id: 'e1', vehicleId: 'v2', category: 'Toll', amount: 45, date: '2026-07-08', notes: 'Highway toll' },
  { id: 'e2', vehicleId: 'v1', category: 'Insurance', amount: 220, date: '2026-07-01' },
  { id: 'e3', vehicleId: 'v3', category: 'Maintenance', amount: 420, date: '2026-07-09', notes: 'Brake pad replacement' },
]

File: backend/app/seed.py
from datetime import datetime

from sqlmodel import Session, select

from app.models import (
    Driver,
    DriverStatus,
    Expense,
    ExpenseCategory,
    FuelLog,
    MaintenanceLog,
    MaintenanceStatus,
    Role,
    Trip,
    TripStatus,
    User,
    Vehicle,
    VehicleStatus,
)

SEED_USERS = [
    User(id="u1", name="Maria Chen", email="manager@transitops.com", password="password123", role=Role.fleet_manager),
    User(id="u2", name="Alex Rivera", email="driver@transitops.com", password="password123", role=Role.driver),
    User(id="u3", name="Sam Okafor", email="safety@transitops.com", password="password123", role=Role.safety_officer),
    User(id="u4", name="Priya Nair", email="finance@transitops.com", password="password123", role=Role.financial_analyst),
]

SEED_VEHICLES = [
    Vehicle(id="v1", registration_number="VAN-05", model="Ford Transit", type="Van", max_load_capacity_kg=500, odometer_km=42150, acquisition_cost=38000, status=VehicleStatus.available, region="North"),
    Vehicle(id="v2", registration_number="TRK-12", model="Isuzu NPR", type="Truck", max_load_capacity_kg=3500, odometer_km=88320, acquisition_cost=62000, status=VehicleStatus.on_trip, region="South"),
    Vehicle(id="v3", registration_number="VAN-08", model="Mercedes Sprinter", type="Van", max_load_capacity_kg=800, odometer_km=15200, acquisition_cost=45000, status=VehicleStatus.in_shop, region="East"),
    Vehicle(id="v4", registration_number="TRK-03", model="Volvo FL", type="Truck", max_load_capacity_kg=5000, odometer_km=121000, acquisition_cost=89000, status=VehicleStatus.retired, region="West"),
    Vehicle(id="v5", registration_number="VAN-11", model="Ford Transit", type="Van", max_load_capacity_kg=500, odometer_km=9800, acquisition_cost=39500, status=VehicleStatus.available, region="North"),
    Vehicle(id="v6", registration_number="TRK-19", model="Isuzu NPR", type="Truck", max_load_capacity_kg=3500, odometer_km=55400, acquisition_cost=63000, status=VehicleStatus.on_trip, region="South"),
    Vehicle(id="v7", registration_number="CAR-02", model="Toyota Corolla", type="Car", max_load_capacity_kg=150, odometer_km=31000, acquisition_cost=21000, status=VehicleStatus.available, region="East"),
]

SEED_DRIVERS = [
    Driver(id="d1", name="Alex Rivera", license_number="LIC-10023", license_category="C", license_expiry="2027-03-15", contact="555-0101", safety_score=92, status=DriverStatus.available),
    Driver(id="d2", name="Jordan Blake", license_number="LIC-10078", license_category="C+E", license_expiry="2026-11-02", contact="555-0102", safety_score=87, status=DriverStatus.on_trip),
    Driver(id="d3", name="Taylor Reed", license_number="LIC-10099", license_category="B", license_expiry="2026-01-20", contact="555-0103", safety_score=78, status=DriverStatus.off_duty),
    Driver(id="d4", name="Morgan Diaz", license_number="LIC-10112", license_category="C", license_expiry="2025-09-01", contact="555-0104", safety_score=65, status=DriverStatus.suspended),
    Driver(id="d5", name="Casey Kim", license_number="LIC-10145", license_category="C+E", license_expiry="2027-06-30", contact="555-0105", safety_score=95, status=DriverStatus.on_trip),
    Driver(id="d6", name="Riley Santos", license_number="LIC-10201", license_category="B", license_expiry="2026-08-14", contact="555-0106", safety_score=89, status=DriverStatus.available),
]

SEED_TRIPS = [
    Trip(id="t1", source="Warehouse A", destination="Downtown Depot", vehicle_id="v2", driver_id="d2", cargo_weight_kg=2800, planned_distance_km=65, status=TripStatus.dispatched, created_at=datetime.fromisoformat("2026-07-10T08:00:00"), dispatched_at=datetime.fromisoformat("2026-07-10T08:15:00")),
    Trip(id="t2", source="Port Terminal", destination="North Storage", vehicle_id="v6", driver_id="d5", cargo_weight_kg=3100, planned_distance_km=40, status=TripStatus.dispatched, created_at=datetime.fromisoformat("2026-07-11T06:30:00"), dispatched_at=datetime.fromisoformat("2026-07-11T07:00:00")),
    Trip(id="t3", source="Warehouse B", destination="Retail Center", vehicle_id="v1", driver_id="d1", cargo_weight_kg=420, planned_distance_km=22, actual_distance_km=23.5, fuel_consumed_liters=6.1, status=TripStatus.completed, created_at=datetime.fromisoformat("2026-07-08T09:00:00"), dispatched_at=datetime.fromisoformat("2026-07-08T09:20:00"), completed_at=datetime.fromisoformat("2026-07-08T11:05:00")),
    Trip(id="t4", source="Depot 5", destination="Customer Site", vehicle_id="v5", driver_id="d6", cargo_weight_kg=300, planned_distance_km=18, status=TripStatus.draft, created_at=datetime.fromisoformat("2026-07-12T07:00:00")),
]

SEED_MAINTENANCE = [
    MaintenanceLog(id="m1", vehicle_id="v3", description="Brake pad replacement", cost=420, odometer_km_at_open=15100, opened_at=datetime.fromisoformat("2026-07-09T10:00:00"), status=MaintenanceStatus.active),
    MaintenanceLog(id="m2", vehicle_id="v1", description="Oil change", cost=90, odometer_km_at_open=32000, opened_at=datetime.fromisoformat("2026-06-15T10:00:00"), closed_at=datetime.fromisoformat("2026-06-15T14:00:00"), status=MaintenanceStatus.closed),
]

SEED_FUEL_LOGS = [
    FuelLog(id="f1", vehicle_id="v1", liters=40, cost=62, date="2026-07-05"),
    FuelLog(id="f2", vehicle_id="v2", liters=120, cost=186, date="2026-07-08"),
    FuelLog(id="f3", vehicle_id="v6", liters=95, cost=148, date="2026-07-09"),
    FuelLog(id="f4", vehicle_id="v1", liters=38, cost=59, date="2026-06-20"),
]

SEED_EXPENSES = [
    Expense(id="e1", vehicle_id="v2", category=ExpenseCategory.toll, amount=45, date="2026-07-08", notes="Highway toll"),
    Expense(id="e2", vehicle_id="v1", category=ExpenseCategory.insurance, amount=220, date="2026-07-01"),
    Expense(id="e3", vehicle_id="v3", category=ExpenseCategory.maintenance, amount=420, date="2026-07-09", notes="Brake pad replacement"),
]


def seed_if_empty(session: Session) -> None:
    if session.exec(select(User)).first():
        return
    for batch in (SEED_USERS, SEED_VEHICLES, SEED_DRIVERS, SEED_TRIPS, SEED_MAINTENANCE, SEED_FUEL_LOGS, SEED_EXPENSES):
        session.add_all(batch)
    session.commit()
